flake.nix

```
@@ -6,56 +6,76 @@
+        "aarch64-darwin"
+        "aarch64-linux"
+      ];
+      forEachSystem = inputs.nixpkgs.lib.genAttrs systems;
+      pkgsFor = forEachSystem (
+        system:
+        import inputs.nixpkgs {
+          inherit system;
+          config.allowUnfree = true;
+        }
+      );
 
-      devShellFor = system: nixpkgs.lib.genAttrs [ "default" ] (_: nixpkgs.legacyPackages.${system}.mkShell {
-        buildInputs = [
-          pkgsFor.${system}.fish
-          pkgsFor.${system}.typescript
-          pkgsFor.${system}.bun
-          pkgsFor.${system}.libnotify
-          pkgsFor.${system}.dart-sass
-          pkgsFor.${system}.fd
-          pkgsFor.${system}.btop
-          pkgsFor.${system}.bluez
-          pkgsFor.${system}.bluez-tools
-          pkgsFor.${system}.grimblast
-          pkgsFor.${system}.gpu-screen-recorder
-          pkgsFor.${system}.brightnessctl
-          pkgsFor.${system}.gnome.gnome-bluetooth
-          pkgsFor.${system}.python3
-          inputs.ags.packages.${system}.agsWithTypes
-        ];
-        nativeBuildInputs = with pkgsFor.${system}; [
-          nixpkgs-fmt
-          nil
-        ];
-        shellHook = ''
-          export GDK_BACKEND=wayland
-        '';
-      });  
-  in {
-    devShells = forEachSystem (system: devShellFor system);
+      devShellFor =
+        system:
+        inputs.nixpkgs.lib.genAttrs [ "default" ] (
+          _:
+          inputs.nixpkgs.legacyPackages.${system}.mkShell {
+            buildInputs = [
+              pkgsFor.${system}.esbuild
+              pkgsFor.${system}.fish
+              pkgsFor.${system}.typescript
+              pkgsFor.${system}.bun
+              pkgsFor.${system}.libnotify
+              pkgsFor.${system}.dart-sass
+              pkgsFor.${system}.fd
+              pkgsFor.${system}.btop
+              pkgsFor.${system}.bluez
+              pkgsFor.${system}.bluez-tools
+              pkgsFor.${system}.grimblast
+              pkgsFor.${system}.gpu-screen-recorder
+              pkgsFor.${system}.brightnessctl
+              pkgsFor.${system}.gnome.gnome-bluetooth
+              pkgsFor.${system}.python3
+              pkgsFor.${system}.matugen
+              inputs.ags.packages.${system}.agsWithTypes
+            ];
+            nativeBuildInputs = with pkgsFor.${system}; [
+              nixfmt-rfc-style
+              nil
+            ];
+            shellHook = ''
+              export GDK_BACKEND=wayland
+            '';
+          }
+        );
+    in
+    {
+      devShells = forEachSystem devShellFor;
 
-    overlay = forEachSystem(system: 
-      let
-        pkgs = pkgsFor.${system};
-      in final: prev: {
-        hyprpanel = (pkgs.callPackage ./. { inherit inputs; }).desktop.script;
-      });
+      overlay = forEachSystem (
+        system:
+        let
+          pkgs = pkgsFor.${system};
+        in
+        final: prev: { hyprpanel = (pkgs.callPackage ./nix { inherit inputs; }).desktop.script; }
+      );
```

Overlays should **never** need to be made per system
```suggestion
    overlay.default = final: prev: {
      hyprpanel = self.packages.${prev.stdenv.system}.default;
    });
```